framework/db/QueryBuilder.php

```diff
@@ -802,6 +802,34 @@ public function dropCommentFromTable($table)
     }
 
     /**
+     * Creates an SQL View.
+     *
+     * @param string $viewName the name of the view to be created.
+     * @param string|Query $subquery the select statement which defines the view.
+     * This can be either a string or a [[Query]] object.
+     * @return string the `CREATE VIEW` SQL statement.
+     */
+    public function createView($viewName, $subquery)
+    {
+        if ($subquery instanceof Query) {
+            list($subquery, $params) = $this->build($subquery);
+            // TODO how to deal with params here?
```

Comment:
TODO should be addressed.

Reply:
in this case its code specifically written by the developer and specifically executed by the developer without anyone else adding anything.

why is it unsafe to use on such conditions?